cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -357,16 +369,28 @@ def get_active_ceph_mgr(self):
             monitoring_endpoint_port = self.DEFAULT_MONITORING_ENDPOINT_PORT
 
         try:
+            failed_ip = monitoring_endpoint_ip
             monitoring_endpoint_ip = self._convert_hostname_to_ip(
                 monitoring_endpoint_ip)
+            # collect all the 'stand-by' mgr ips
+            mgr_ips = []
+            for each_standby_mgr in standby_mgrs:
+                failed_ip = each_standby_mgr
+                mgr_ips.append(
+                        self._convert_hostname_to_ip(each_standby_mgr))
         except:
             raise ExecutionFailureException(
-                "unable to convert a hostname to an IP address, monitoring host name: {}".format(monitoring_endpoint_ip))
+                "Conversion of host: {} to IP failed. "\
```

Comment:
Are you returning an error here? The CI fails but it looks like the script outputs 0. See: 

```
2s
Run kubectl -n rook-ceph exec $(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[0].metadata.name}') -- mkdir -p /etc/ceph/test-data
  kubectl -n rook-ceph exec $(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[0].metadata.name}') -- mkdir -p /etc/ceph/test-data
  kubectl -n rook-ceph cp cluster/examples/kubernetes/ceph/test-data/ceph-status-out $(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[0].metadata.name}'):/etc/ceph/test-data/
  kubectl -n rook-ceph cp cluster/examples/kubernetes/ceph/create-external-cluster-resources.py $(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[0].metadata.name}'):/etc/ceph
  kubectl -n rook-ceph exec $(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[0].metadata.name}') -- python3 /etc/ceph/create-external-cluster-resources.py --rbd-data-pool-name replicapool
  shell: /bin/bash -e {0}
  env:
    GOROOT: /opt/hostedtoolcache/go/1.15.8/x64
    MINIKUBE_HOME: /home/runner/work/_temp
Execution Failed: Conversion of host: rook-ceph-mgr-a-664bb58559-vdwqh to IP failed. Please enter the IP addresses of all the ceph-mgrs with the '--monitoring-endpoint' flag
```

Reply:
I am raising a python exception here, but at the call-site, we are catching the exception thus returning with a ZERO exit value.
Fixed the issue in the latest CI fix commit.

Reply:
What's the fix? Where is it?

Reply:
At the above mentioned line, we are raising the error but at the calling place we are catching/handling the error. So it is terminating successfully. The fix was to re-raise the exception at the call site.
https://github.com/rook/rook/blob/bc4dd243069f5d3af7b04fb07ff489ece30ac78c/cluster/examples/kubernetes/ceph/create-external-cluster-resources.py#L800